src/smart-components/role/add-role/inventory-groups-role.js

```diff
@@ -118,60 +130,71 @@ const InventoryGroupsRole = (props) => {
+            <FormGroup label={permission.replace('inventory:', '')} isRequired />
+          </GridItem>
+          <GridItem lg={7} md={4} sm={8}>
+            <Tooltip content={<div>{intl.formatMessage(messages.inventoryGroupsTooltip)}</div>}>
+              <Select
+                className="rbac-m-cost-resource-select"
+                variant={SelectVariant.checkbox}
+                typeAheadAriaLabel={intl.formatMessage(messages.inventoryGroupsTypeAheadLabel)}
+                aria-labelledby={permission}
+                selections={state[permission].selected.map(({ name }) => name)}
+                placeholderText={intl.formatMessage(messages.selectGroups)}
+                onSelect={(event, selection) =>
+                  onSelect(event, selection, selection === intl.formatMessage(messages.selectAll, { length: options?.length ?? 0 }), permission)
+                }
+                onToggle={(isOpen) => {
+                  dispatchLocally({ type: 'setFilter', key: permission, filterValue: '' });
+                  onToggle(permission, isOpen);
+                }}
+                onClear={() => clearSelection(permission)}
+                onFilter={(e) => e && dispatchLocally({ type: 'setFilter', key: permission, filterValue: e.target.value })}
+                isOpen={state[permission].isOpen}
+                hasInlineFilter
+              >
+                {[
+                  ...(options?.length > 0
+                    ? [<SelectOption key={`${permission}-all`} value={intl.formatMessage(messages.selectAll, { length: options?.length })} />]
+                    : []),
+                  ...(options?.map((option, index) => <SelectOption key={`${permission}-${index + 1}`} value={option?.name} />) || []),
+                ]}
+              </Select>
+            </Tooltip>
+          </GridItem>
+          <GridItem lg={2} md={4} sm={2} className="rbac-m-hide-on-sm">
+            {
+              index <= 0 && (
+                <Button key={index} variant="link" isInLink onClick={() => dispatchLocally({ type: "copyToAll", permissions })}>
+                  Copy to all
+                </Button>
+              )
+            }
+          </GridItem>
+        </Grid>
       </React.Fragment>
     );
   };
 
   return (
     <Grid hasGutter>
-      <GridItem md={4} className="rbac-m-hide-on-sm">
+      <GridItem lg={3} md={6} className="rbac-m-hide-on-sm">
         <Text component={TextVariants.h4} className="rbac-bold-text pf-u-mt-sm">
           {intl.formatMessage(messages.permissions)}
         </Text>
       </GridItem>
-      <GridItem md={8} className="rbac-m-hide-on-sm">
+      <GridItem lg={9} md={6} className="rbac-m-hide-on-sm">
         <Text component={TextVariants.h4} className="rbac-bold-text pf-u-mt-sm">
           {intl.formatMessage(messages.groupDefinition)}
         </Text>
       </GridItem>
-      {permissions.map(makeRow)}
+      {permissions.map((permission, index) => makeRow(permission, index))}
```

Comment:
This change is not required. The arguments have the same order and values.

Reply:
Only way I was able to successfully grab the index inside of the make row function, leaving it implicit was only making it complaint about missing definition for it.

Reply:
You were right 😅 

I was just messing up how i was making the call originally, all should be gucci now